libethash/internal.c

```diff
@@ -208,7 +219,14 @@ static bool ethash_hash(
 		for (unsigned n = 0; n != MIX_NODES; ++n) {
 			node const* dag_node;
 			if (full_nodes) {
+#if defined(__MIC__)
+				// TODO: fix dag_node
+				node tmp_node;
+				tmp_node = full_nodes[MIX_NODES * index + n];
+				dag_node = &tmp_node;
+#else
 				dag_node = &full_nodes[MIX_NODES * index + n];
```

Comment:
@bobsummerwill @chriseth 
Hello,

Any idea where memory allocation for the full_nodes object occurs?
It seems that the memory is not aligned for use with m512i.
The code produces a Segmentation Fault without a tmp_node copy.
A Segmentation Fault usually indicates that the memory is not aligned.
 
Thanks,
coast